stdlib/public/TensorFlow/Ops.swift

```diff
@@ -1597,52 +1597,257 @@ public extension Tensor where Scalar : Numeric {
+        ellipsisMask: indexPath.ellipsisMask, newAxisMask: indexPath.newAxisMask, 
+        shrinkAxisMask: indexPath.squeezeAxisMask)
+    }
+    set {
+      self = Raw.tensorStridedSliceUpdate(
+        self, begin: indexPath.begin, end: indexPath.end, strides: indexPath.strides,
+        value: newValue, beginMask: indexPath.beginMask, endMask: indexPath.endMask, 
+        ellipsisMask: indexPath.ellipsisMask, newAxisMask: indexPath.newAxisMask, 
+        shrinkAxisMask: indexPath.squeezeAxisMask)
+    }
+  }
+
+  @inlinable @inline(__always)
+  subscript(_ indices: TensorSliceIndexProtocol...) -> Tensor {
+    get {
+      return self[IndexPath(indices.map { $0.sliceIndex })]
+    }
+    set {
+      self[IndexPath(indices.map { $0.sliceIndex })] = newValue
+    }
+  }
+
+  @inlinable @inline(__always)
+  internal func _vjpSubscript(_ indexPath: IndexPath) -> (Tensor, (Tensor) -> Tensor) {
+    return (self[indexPath], { [shape = shapeTensor] v in
+      return Tensor<Scalar>._pullbackSubscript(v, indexPath, shape)
+    })
+  }
+
+  @inlinable @inline(__always)
+  @differentiable(wrt: seed, vjp: _vjpPullbackSubscript)
+  internal static func _pullbackSubscript(
+    _ seed: Tensor,
+    _ indexPath: IndexPath,
+    _ shape: Tensor<Int32>
+  ) -> Tensor {
+    return Raw.stridedSliceGrad(
+      shape: shape, begin: indexPath.begin, end: indexPath.end, strides: indexPath.strides,
+      dy: seed, beginMask: indexPath.beginMask, endMask: indexPath.endMask,
+      ellipsisMask: indexPath.ellipsisMask, newAxisMask: indexPath.newAxisMask,
+      shrinkAxisMask: indexPath.squeezeAxisMask)
+  }
+
+  @inlinable @inline(__always)
+  internal static func _vjpPullbackSubscript(
+    _ seed: Tensor,
+    _ indexPath: IndexPath,
+    shape: Tensor<Int32>
+  ) -> (Tensor, (Tensor) -> Tensor) {
+    return (Tensor<Scalar>._pullbackSubscript(seed, indexPath, shape), { v in
+      return v[IndexPath(
+        begin: indexPath.begin, end: indexPath.end, strides: indexPath.strides,
+        beginMask: indexPath.beginMask, endMask: indexPath.endMask,
+        ellipsisMask: indexPath.ellipsisMask, newAxisMask: indexPath.newAxisMask,
+        squeezeAxisMask: indexPath.squeezeAxisMask)]
+    })
+  }
+}
+
+public extension Tensor.IndexPath {
+  @inlinable @inline(__always)
+  init(_ indices: [TensorSliceIndex]) {
+    precondition(!indices.isEmpty, "The index path cannot be empty.")
+    precondition(indices.count(where: {
+      if case .ellipsis = $0 {
+        return true
+      } else {
+        return false
+      }
+    }) < 2, "Only one ellipsis is allowed per index path.")
+
+    var begin = [Int32](repeating: 0, count: indices.count)
+    var end = [Int32](repeating: 0, count: indices.count)
+    var strides = [Int32](repeating: 1, count: indices.count)
+    var beginMask: Int64 = 0
+    var endMask: Int64 = 0
+    var ellipsisMask: Int64 = 0
+    var newAxisMask: Int64 = 0
+    var squeezeAxisMask: Int64 = 0
+    for (i, index) in indices.enumerated() {
```

Comment:
I'm worried about cases where number of indices is greater than 64. It's not common, but it would be ideal to make sure every quantity in our API is bound by `Int32` or `Int`, not relatively small numbers.

Reply:
I'm not sure if TF core supports tensors with rank higher than 64 actually. The only reason I use Int64 here is that that's what's specified in the strided slice op in TF core.

Reply:
Given that these are all internal now, can I leave this as is for now?

Reply:
Yep, no problem.